carp.deployment.core/src/commonMain/kotlin/dk/cachet/carp/deployment/domain/DeviceDeploymentStatus.kt

```diff
@@ -9,31 +9,68 @@ import kotlinx.serialization.Serializable
  * Describes the status of a device, part of a study deployment.
  */
 @Serializable
-data class DeviceDeploymentStatus(
+sealed class DeviceDeploymentStatus
+{
     /**
      * The description of the device.
      */
     @Serializable( DeviceDescriptorSerializer::class )
-    val device: AnyDeviceDescriptor,
+    abstract val device: AnyDeviceDescriptor
     /**
      * Determines whether registering the device is required for the deployment to start running.
      */
-    val requiresRegistration: Boolean,
-    /**
-     * Determines whether the device is currently registered.
-     */
-    val isRegistered: Boolean,
+    abstract val requiresRegistration: Boolean
     /**
      * Determines whether the device requires a device deployment by retrieving [MasterDeviceDeployment].
      * Not all master devices necessarily need deployment; chained master devices do not.
      */
-    val requiresDeployment: Boolean,
+    abstract val requiresDeployment: Boolean
+
+
+    /**
+     * A device deployment status which has not been deployed yet.
+     */
+    interface NotDeployed
+    {
+        /**
+         * Determines whether the device has been registered successfully and is ready for deployment.
+         */
+        val isReadyForDeployment: Boolean
+    }
+
+
+    /**
+     * Device deployment status for when a device has not been registered.
+     */
+    @Serializable
+    data class Unregistered(
+        @Serializable( DeviceDescriptorSerializer::class )
+        override val device: AnyDeviceDescriptor,
+        override val requiresRegistration: Boolean,
+        override val requiresDeployment: Boolean,
+        override val isReadyForDeployment: Boolean
```

Comment:
Yes, it should be always false. I considered for some time whether there would be use cases where this logic in `StudyDeployment` changes, and this would decouple that, but given our discussion today in terms of using `deviceId` to determine succesful deployment, we know _registration will always be needed_.

Thus, I will also reflect that in this data class by removing it from the constructor and moving it to a default initialized field. However, I do still think it is useful for the field to be present to implement the UI (study manager), otherwise the domain logic that an unregistered device can never be deployed leaks all through the front-end code.